### ISSUE TITLE
[🚀 Feature]: Be able to provide custom path to Safari/Safari Technology Preview binaries

### ISSUE THREAD

**Issue by 92kns** [SeleniumHQ/selenium#13714]
### Feature and motivation

I would like to provide custom paths to  Safari/Safari Technology Preview/(maybe even webkit binaries) for some automation, similar to how you can do that for other browsers e.g. [firefox](https://github.com/SeleniumHQ/selenium/blob/864089606c3780574630d28c0173cb567516b8cf/javascript/node/selenium-webdriver/firefox.js), [chrome](https://github.com/SeleniumHQ/selenium/blob/864089606c3780574630d28c0173cb567516b8cf/javascript/node/selenium-webdriver/chrome.js) etc

Right now the assumption is that the binary is always in the `/Application/<...>` folder (hard coded?)

What would be the way forward for this?

### Usage example

Basically, the same way one can set paths in [firefox](https://github.com/SeleniumHQ/selenium/blob/864089606c3780574630d28c0173cb567516b8cf/javascript/node/selenium-webdriver/firefox.js), [chrome](https://github.com/SeleniumHQ/selenium/blob/864089606c3780574630d28c0173cb567516b8cf/javascript/node/selenium-webdriver/chrome.js), i was hoping to have a similar method with safari/safari-tp in [safari](https://github.com/SeleniumHQ/selenium/blob/864089606c3780574630d28c0173cb567516b8cf/javascript/node/selenium-webdriver/safari.js)

**Comment by diemol** [comment]
How can you tell SafariDriver to use a custom path? Is that possible?

**Comment by 92kns** [comment]
@diemol thanks for getting back to me. I am not aware of a way - so does this mean safaridriver is the actual blocker here from preventing this feature?

**Comment by diemol** [comment]
I believe so.
